python/adjoint/objective.py

```diff
@@ -21,9 +41,75 @@ def register_monitors(self, frequencies):
     def place_adjoint_source(self, dJ):
         """Places appropriate sources for the adjoint simulation."""
 
-    @abc.abstractmethod
     def get_evaluation(self):
         """Evaluates the objective quantity."""
+        if self._eval:
+            return self._eval
+        else:
+            raise RuntimeError(
+                'You must first run a forward simulation before resquesting the evaluation of an `ObjectiveQuantity`.'
+            )
+
+    def _adj_src_scale(self, include_resolution=True):
+        """Calculates the scale for the adjoint sources."""
+        T = self.sim.meep_time()
+        dt = self.sim.fields.dt
+        src = self._create_time_profile()
+
+        if include_resolution:
+            num_dims = ((self.sim.cell_size.x != 0) +
+                        (self.sim.cell_size.y != 0) +
+                        (self.sim.cell_size.z != 0))
```

Comment:
Meep only supports 2d simulations in the xy plane, for example — if you want to do an xz plane, you do it as a 3d simulation that is 1 pixel thick and periodic in x. 

I feel like this should use `_infer_dimensions`.   (Note that cylindrical coordinates are special, but Mo has code to handle that.)